fission/src/ui/panels/configuring/MatchModeConfigPanel.tsx

```diff
@@ -0,0 +1,286 @@
+            </Box>
+        </Box>
+    )
+}
+
+const MatchModeConfigPanel: React.FC<PanelPropsImpl> = ({ panelId }) => {
+    const { closePanel } = usePanelControlContext()
+    const { openModal } = useModalControlContext()
+
+    const [matchModeConfigs, setMatchModeConfigs] = useState<MatchModeConfig[]>([])
+
+    useEffect(() => {
+        const loadAllJsonFiles = async () => {
+            try {
+                const indexRes = await fetch("match-mode-config/index.json")
+                const fileNames: string[] = await indexRes.json()
+
+                const defaultConfigs = await Promise.all(
+                    fileNames.map(async fileName => {
+                        const res = await fetch(`match-mode-config/${fileName}`)
+                        return res.json()
+                    })
+                )
+                const localConfigs = JSON.parse(window.localStorage.getItem("match-mode-configs") || "[]")
+
+                const combinedConfigs = [...defaultConfigs, ...localConfigs]
+                const uniqueConfigsById = Array.from(new Map(combinedConfigs.map(item => [item.id, item])).values())
+
+                setMatchModeConfigs(uniqueConfigsById)
+            } catch (err) {
+                console.error("Error loading JSON files:", err)
+                Global_AddToast?.(
+                    "error",
+                    "Error Loading Match Mode Configs",
+                    "Please check the console for more details"
+                )
+            }
+        }
+
+        loadAllJsonFiles()
+    }, [])
+
+    const matchModeConfigElements = useMemo(
+        () =>
+            matchModeConfigs.map(config =>
+                ItemCard({
+                    id: config.id,
+                    name: config.name || config.id || "Unnamed Match Mode",
+                    primaryOnClick: () => {
+                        MatchConfigSelected(config, openModal)
+                        closePanel("match-mode-config")
+                    },
+                    secondaryOnClick: () => {
+                        // Delete the config from the local storage
+                        setMatchModeConfigs(prev => prev.filter(c => c.id !== config.id))
+                        window.localStorage.setItem(
+                            "match-mode-configs",
+                            JSON.stringify(matchModeConfigs.filter(c => c.id !== config.id))
+                        )
+                        matchModeConfigs.filter(c => c.id !== config.id)
+                    },
+                })
+            ),
+        [matchModeConfigs, openModal, closePanel]
+    )
+
+    const fileUploadRef = useRef<HTMLInputElement>(null)
+
+    const uploadClicked = () => {
+        if (fileUploadRef.current) {
+            fileUploadRef.current.click()
+        }
+    }
+
+    const validateMatchModeConfig = (config: unknown): config is MatchModeConfig => {
```

Comment:
Can we not just make `MatchModeConfig` the type of `config` directly? Like we can parse the object to a specific type and just catch that, bubbling the parse error back to the user. This feels too complicated for what we need. Additionally, if we're doing this whole validation function, we can assume `config` is of type `any` at least, since we can default an empty file to an empty object.

There're some things I really like about the way you handled this, and some things that could be done better. I would love to sit down with you some time and go over some of the nuances of types and json parsing in TS.